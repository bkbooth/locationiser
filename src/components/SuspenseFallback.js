import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSpinnerThird } from '@fortawesome/pro-solid-svg-icons';
import PageWrapper from './PageWrapper';

function SuspenseFallback() {
  return (
    <PageWrapper>
      <p>
        <FontAwesomeIcon icon={faSpinnerThird} spin={true} /> Loading...
      </p>
    </PageWrapper>
  );
}

export default SuspenseFallback;
