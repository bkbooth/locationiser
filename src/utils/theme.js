export const theme = {
  colours: {
    primary: {
      '500': '#0094c0',
      '600': '#007192',
    },
    shade: {
      '100': '#101010',
      '400': '#575757',
      '500': '#656565',
      '600': '#929292',
      '800': '#ececec',
      '900': '#fbfbfb',
    },
    positive: {
      '500': '#38c000',
      '600': '#2c9600',
    },
    negative: {
      '500': '#c00000',
    },
  },
  fonts: {
    sans: `-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif`,
  },
  sizes: {
    nil: '0',
    xxs: '1px',
    xs: '5px',
    sm: '10px',
    md: '20px',
    lg: '30px',
    xl: '40px',
  },
  speeds: {
    fast: '0.2s',
    slow: '0.5s',
  },
};
