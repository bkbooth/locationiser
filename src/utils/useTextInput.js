import { useState } from 'react';

export function useTextInput(initialValue) {
  const [value, setValue] = useState(initialValue);

  function onChange(event) {
    setValue(event.currentTarget.value);
  }

  return { value, onChange };
}
