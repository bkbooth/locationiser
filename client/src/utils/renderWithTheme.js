import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components/macro';
import { theme } from 'utils/theme';

function WithThemeProvider({ children }) {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
}

export function renderWithTheme(ui, options) {
  return render(ui, { wrapper: WithThemeProvider, ...options });
}
