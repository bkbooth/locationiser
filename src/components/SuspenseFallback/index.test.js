import React from 'react';
import { renderWithTheme } from 'utils/renderWithTheme';
import SuspenseFallback from './index';

it('renders correctly', () => {
  const { container } = renderWithTheme(<SuspenseFallback />);
  expect(container.cloneNode(true)).toMatchSnapshot();
});
