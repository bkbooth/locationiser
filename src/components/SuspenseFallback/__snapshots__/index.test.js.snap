// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 100%;
  max-height: calc(100% - 2 * 20px);
  overflow-y: auto;
  margin: 20px;
  border-radius: 5px;
  background: #fbfbfb;
  box-shadow: 0 10px 40px rgba(0,0,0,0.3);
}

.c4 {
  padding: 30px 20px;
}

.c4 *:first-child {
  margin-top: 0;
}

.c4 *:last-child {
  margin-bottom: 0;
}

.c1 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  left: 0;
  right: 0;
  height: 70px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 20px;
  background: #0094c0;
  color: #fbfbfb;
  box-shadow: 0 5px 10px rgba(0,0,0,0.4);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
  font-size: 1.7rem;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
}

.c3 {
  width: 40px;
  height: auto;
  margin-right: 10px;
}

@media screen and (min-width:540px) {
  .c0 {
    width: 500px;
    margin: 20px auto;
  }
}

<div>
  <div
    class="c0"
  >
    <header
      class="c1"
    >
      <h1
        class="c2"
      >
        <img
          alt="world map emoji"
          class="c3"
          src="world-map-emoji.png"
        />
        locations
      </h1>
    </header>
    <div
      class="c4"
    >
      <p>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-spinner-third fa-w-16 fa-spin "
          data-icon="spinner-third"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M456.433 371.72l-27.79-16.045c-7.192-4.152-10.052-13.136-6.487-20.636 25.82-54.328 23.566-118.602-6.768-171.03-30.265-52.529-84.802-86.621-144.76-91.424C262.35 71.922 256 64.953 256 56.649V24.56c0-9.31 7.916-16.609 17.204-15.96 81.795 5.717 156.412 51.902 197.611 123.408 41.301 71.385 43.99 159.096 8.042 232.792-4.082 8.369-14.361 11.575-22.424 6.92z"
            fill="currentColor"
          />
        </svg>
         Loading...
      </p>
    </div>
  </div>
</div>
`;
